src/pipx/interpreter.py

```diff
@@ -56,7 +57,23 @@ def _find_default_windows_python() -> str:
     return python  # This executable seems to work.
 
 
-if WINDOWS:
-    DEFAULT_PYTHON = _find_default_windows_python()
+def _get_sys_executable() -> str:
+    if WINDOWS:
+        return _find_default_windows_python()
+    else:
+        return sys.executable
+
+
+def _get_absolute_python_interpreter(env_python: str) -> str:
+    which_python = shutil.which(env_python)
+    if not which_python:
+        raise PipxError(f"Default python interpreter {repr(env_python)} is invalid")
+    return which_python
+
+
+env_default_python = os.environ.get("PIPX_DEFAULT_PYTHON")
```

Comment:
Since `env_default_python` is not used anywhere else but this `if` statement, could you combine lines 74,76 into one line for code simplicity and readability:
```
if not os.environ.get("PIPX_DEFAULT_PYTHON"):
```

Reply:
It's used in two places below, which is why I assigned it to a variable instead of calling `os.environ` twice. Do you still want me to get rid of it?

Reply:
Oh sorry I see inside the else.  Ok you can forget this.